js/interactive-guides/microprofile-config/microprofile-config-callback.js

```diff
@@ -267,17 +294,108 @@ var microprofileConfigCallBack = (function() {
         editor.addSaveListener(__showWebBrowser);
     };
 
+    var __getMicroProfileConfigFeatureContent = function(content) {
+      var editorContents = {};
+      try {
+          // match
+          // <feature>jaxrs-2.0</feature>
+          //    <anything here>
+          // </featureManager>
+          // and capturing groups to get content before <feature>jaxrs-2.0</feature>, the feature, and after
+          // closing featureManager content tag.
+          var featureManagerToMatch = "([\\s\\S]*)<feature>jaxrs-2.0</feature>([\\s\\S]*)<\\/featureManager>([\\s\\S]*)";
+          var regExpToMatch = new RegExp(featureManagerToMatch, "g");
+          var groups = regExpToMatch.exec(content);
+          editorContents.beforeNewFeature = groups[1]; //includes <feature>jaxrs-2.0</feature>
+          editorContents.features = groups[2];
+          editorContents.afterFeature = groups[3];
+      }
+      catch (e) {
+
+      }
+      return editorContents;
+    };
+
+
+    var __isConfigInFeatures = function(features) {
+       var match = false;
+       features = features.replace('\n', '');
+       features = features.replace(/\s/g, ''); // Remove whitespace
+       try {
+           var featureMatches = features.match(/<feature>[\s\S]*?<\/feature>/g);
+           $(featureMatches).each(function (index, feature) {
+               if (feature.indexOf("<feature>mpConfig-1.1</feature>") !== -1) {
+                   match = true;
+                   return false; // break out of each loop
+               }
+
+           });
+       }
+       catch (e) {
+
+       }
+       return match;
+   };
+
+  var __checkMicroProfileConfigFeatureContent = function(content) {
+      var isConfigFeatureThere = true;
+      var editorContentBreakdown = __getMicroProfileConfigFeatureContent(content);
+      if (editorContentBreakdown.hasOwnProperty("features")) {
+        isConfigFeatureThere =  __isConfigInFeatures(editorContentBreakdown.features);
+        if (isConfigFeatureThere) {
+              // check for whether other stuffs are there
+              var features = editorContentBreakdown.features;
```

Comment:
It seems like __isConfigInFeatures could take care of the checking. There is no need for two levels of checking.